source/parser.hpp

```diff
@@ -1,25 +1,40 @@
 #pragma once
 
-#include "../source/simulator.hpp"
+#include <yaml-cpp/yaml.h>
+
+#include <filesystem>
 #include <map>
 #include <string>
-#include <cstdint>
-#include <yaml-cpp/yaml.h>
 
+#include "simulator.hpp"
 
 namespace sim {
-	class YamlParser {
-	public:
-		BasicSimulator parseConfig(const std::string& filename);
 
-	private:
-		std::map<std::string, std::shared_ptr<IRoutingDevice>> devices_map;
+class YamlParser {
+public:
+    SimulatorVariant buildSimulatorFromConfig(
+        const std::filesystem::path& path);
+    Time get_simulation_time() const;
```

Comment:
1) Please, use camel case in name of methods (`buildSimulatorFromConfig`)
2) I think it would be more clear to return both simulator and simulation time from parsing function: you may call `get_simulatio_time` before `buildSimulatorFromConfig`, what is strange
```suggestion
    std::pair<SimulatorVariant, Time> parse_simulation_config(
        const std::filesystem::path& path);
```